src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py

```diff
@@ -370,6 +370,27 @@ def test_aks_create_with_openservicemesh_addon(self, resource_group, resource_gr
             self.check('addonProfiles.openServiceMesh.enabled', True),
         ])
 
+    @AllowLargeResponse()
+    @AKSCustomResourceGroupPreparer(random_name_length=17, name_prefix='clitest', location='westus2')
+    def test_aks_addon_list(self, resource_group, resource_group_location):
+        print("what up?")
+        aks_name = self.create_random_name('cliakstest', 16)
+        self.kwargs.update({
+            'resource_group': resource_group,
+            'name': aks_name,
+        })
+
+        create_cmd = 'aks create --resource-group={resource_group} --name={name} --enable-managed-identity --generate-ssh-keys -o json'
+        self.cmd(create_cmd, checks=[
+            self.check('provisioningState', 'Succeeded'),
+            self.check('addonProfiles.openServiceMesh', None),
+        ])
+
+        list_cmd = 'aks addon list --resource-group={resource_group} --name={name} -o json'
+        addon_list = self.cmd(list_cmd).get_output_in_json()
```

Comment:
Please use the method provided by testsdk to check the result, this could be something like ```self.check('length(@) == `2`', True)```.

Reply:
Would `@` point to the list I receive from the command?

Reply:
Also the docs recommend using this way of testing for dynamic stuff: https://github.com/Azure/azure-cli/blob/dev/doc/authoring_tests.md#sample-2-validate-the-return-value-in-json

Reply:
You can check the output with JMESPath or just get it done with python. In my opinion, it is more concise to use JMESPath when the logic is not complicated. Back to this case, it's all good to check with `assert`.